### PR TITLE
Refactor extractor

### PR DESCRIPTION
**Pull request by wichersn**

### CODE REVIEW

**Comment by wichersn** [comment]
Refactored to use Extractor base class for averaging multiple training runs in both tf and torch.

**Comment by ricvolpe** [comment]
Since we are refactoring, does it make sense as part of this pull request to delete
* `extract_preferences.ipynb`
* `find_subnets_torch.ipynb`
from the root of the repo?

File: extractors/configs/hptuning_config_torch.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 trainingInput:
   scaleTier: CUSTOM
-  masterType: n1-highmem-4
+  masterType: n1-standard-4
   hyperparameters:
     goal: MAXIMIZE
     hyperparameterMetricTag: mean_val_auc
     maxTrials: 60
-    maxParallelTrials: 4
+    maxParallelTrials: 5
```
Comment (ricvolpe) [inline extractors/configs/hptuning_config_torch.yaml:8]: Should we change the same param also in `extractors/configs/hptuning_config_tf.yaml` then?

File: extractors/configs/tf.gin [hunk 2]
```diff
@@ -8,10 +8,12 @@ data_pipeline.rebalance = True
 
 extractor_type.extractor = @TfExtractor
 
+extractor.train.do_summary = True
+Extractor.num_repeat = 10
+
 TfExtractor.extractor_fn = @cnn_from_obs
```
Comment (ricvolpe) [inline extractors/configs/tf.gin:14]: Change this to `@agent_extractor` as default - Setting the `extractor_fn` to the CNN from obs is already taken care of in `extractors/configs/hptuning_config_tf_baseline.yaml` (line 13)

File: extractors/extractor.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        metrics = {
+            "mean_val_auc": np.mean(all_val_auc),
+            "mean_val_accuracy": np.mean(all_val_accuracy),
+            "val_auc_std": np.std(all_val_auc),
+            "val_accuracy_std": np.std(all_val_accuracy)
+        }
+
+        if do_summary:
+            print(metrics, flush=True)
```
Comment (ricvolpe) [inline extractors/extractor.py:55]: Could it happen that we also want to print train metrics is the future? For example we want to see if an agent can learn the preferences on the train set first on a new enviroment (debug model capacity vs overfitting)

File: extractors/configs/tf.gin (hunk 2 above)
Comment (ricvolpe) [reply extractors/configs/tf.gin:14]: Nevermind - I can see them in the logs.

**Review by ricvolpe — APPROVED** [review]

File: extractors/extractor.py (hunk 3 above)
Comment (ricvolpe) [reply extractors/extractor.py:55]: Nevermind - I can see them in the logs.

File: extractors/configs/tf.gin (hunk 2 above)
Comment (ricvolpe) [reply extractors/configs/tf.gin:14]: Saw you changed this in the next PR - ok.

File: extractors/configs/hptuning_config_torch.yaml (hunk 1 above)
Comment (wichersn) [reply extractors/configs/hptuning_config_torch.yaml:8]: Yeah, I did that in the next pr.

**Pull request merged**